### DISCUSSION TITLE
Modify block pallete blocks.

### DISCUSSION THREAD

**Discussion by yuri-kiss** [TurboWarp/extensions#646] [Q&A]
I want to modify the block palette to add more blocks to the existing category's, without modifying the scratch-vm.

**Comment by yuri-kiss** [comment]
I have tried 
`vm.runtime.ext_scratch3_motion.runtime.addAddonBlock({arguments:[],procedureCode:'uhh test block but for some reason it did not work lol',callback:(function(args){return 754}),hidden:false});
vm.runtime.extensionManager.refreshBlocks();`
but that just adds it to the addons section.
![image](https://github.com/TurboWarp/extensions/assets/135030944/bafab5ba-4586-4d54-8c19-8d689679ddd4)

**Comment by LilyMakesThings** [comment]
If you're referring to the fact way I did it, it's really not ideal. It's why I've not finished TurboCharged.

The only way you're able to do it is by forcing the XML of the Blockly toolbox for the category to be something else, which on its own is fine, but as soon as more than one extension starts doing it, they clash and overwrite eachother.

The only thing I've used the code that replaces toolbox XML for is All Menus, because it's really experimental. The code to do it was originally by Xeltalliv, but I had to make some modifications https://github.com/TurboWarp/extensions/pull/510

**Reply by yuri-kiss** [reply]
its not going to be an extension tbh, Just a personal project,

**Reply by yuri-kiss** [reply]
1. are you manually doing every block that way? 
2. did you create some helper function to semi-automate it, if so how.
3. Do you plan to release TurboCharged if you finish it?

**Reply by LilyMakesThings** [reply]
> 1. are you manually doing every block that way?
> 2. did you create some helper function to semi-automate it, if so how.
> 3. Do you plan to release TurboCharged if you finish it?

1. Elaborate please
2. Elaborate please
3. Yes

**Reply by yuri-kiss** [reply]
F1: 
> The only way you're able to do it is by forcing the XML of the Blockly toolbox for the category to be something else, which on its own is fine, but as soon as more than one extension starts doing it, they clash and overwrite eachother.

1. Are you manually doing this for every block (F1).
2. Or did you create a helper function that would speed up the process.

helper function
> A function to help with the process of developing code, Like utilities and alike, example: `GetSpriteByName(name)`
would be shorter than the alternative, so it was put in a function.

**Reply by yuri-kiss** [reply]
Elaborates for you, (did i use that word right 🤔)

**Reply by LilyMakesThings** [reply]
> Elaborates for you, (did i use that word right 🤔)

I think you did :'D 

I'm not doing this for every single block, no, I'm doing this for every category. One time for motion, one time for operators, etc. If you send me your discord I can DM you TurboCharged and you can see how it works there instead

**Reply by yuri-kiss** [reply]
uhh I guess time to make a temporary discord account.

**Reply by yuri-kiss** [reply]
SurvExE-Pc#6483

**Reply by yuri-kiss** [reply]
made it as fast as possible
